lib/Sema/TypeCheckConcurrency.cpp

```diff
@@ -4413,6 +4413,43 @@ bool swift::checkSendableConformance(
   return checkSendableInstanceStorage(nominal, conformanceDC, check);
 }
 
+/// Add "unavailable" attributes to the given extension.
+static void addUnavailableAttrs(ExtensionDecl *ext, NominalTypeDecl *nominal) {
+  ASTContext &ctx = nominal->getASTContext();
+  llvm::VersionTuple noVersion;
+
+  // Add platform-version-specific @available attributes.
+  for (auto available : nominal->getAttrs().getAttributes<AvailableAttr>()) {
+    if (available->Platform == PlatformKind::none)
+      continue;
+
+    auto attr = new (ctx) AvailableAttr(
+        SourceLoc(), SourceRange(),
+        available->Platform,
+        available->Message,
+        "", nullptr,
+        available->Introduced.getValueOr(noVersion), SourceRange(),
+        available->Deprecated.getValueOr(noVersion), SourceRange(),
+        available->Obsoleted.getValueOr(noVersion), SourceRange(),
+        PlatformAgnosticAvailabilityKind::Unavailable,
+        /*implicit=*/true,
+        available->IsSPI);
```

Comment:
I could parameterize clone to allow me to override the `PlatformAgnosticAvailabilityKind::Unavailable` and `Renamed` information, but that felt like overall it would be less clear what "clone" meant and wouldn't remove that much repetition.